--- osu_news.module.old	2012-08-30 14:36:50.000883000 -0700
+++ osu_news.module	2012-08-30 14:47:34.000655000 -0700
@@ -37,6 +37,7 @@
	$html = new DOMDocument; // Need this to parse the description

  foreach ($xml->channel->item as $story) {
+	unset ($date_text);
    if (++$items > $num_items) break;

		unset($teaser); // Don't reuse content from previous iteration
@@ -68,6 +69,9 @@
       	$body = mb_convert_encoding($div->getElementsByTagName('div')->item(2)->nodeValue, "ISO_8859-1", "UTF-8");
	    }
		}
+    if (!$date_text) {
+    	$date_text = $story->pubDate;
+    }
    $content .=  '<div class="news-item">';
    $content .= $thumb;
    $content .= '<h3 class="title"><a href="' . $story->link . '">' . $story->title . '</a></h3> - ';
